### ISSUE TITLE
Javascript array.indexof not implemented

### ISSUE THREAD

**Issue by groogiam** [bridgedotnet/Bridge#3710]
A description of the issue.

### Steps To Reproduce

https://deck.net/ab7ff51288813d22d04dae7b7948847f

### Expected Result
Should be able to call indexof and pass in type of .NET array.

**Comment by fabriciomurta** [comment]
Hello @groogiam!

I've tried your code and it does not run in C#, this *IndexOf()* implementation and usage is not available from C#. or should it work? At least it doesn't [in dotnet fiddle](https://dotnetfiddle.net/LO1hg5).

If this helps, Bridge implements C# interface to draw JavaScript files. Some libraries and extensions that are specific to javascript can also be ported to C# and used just fine (like the ~2000 from retyped). So maybe you're just missing a javascript interface or something like this to implement the javascript-specific behavior into C#?

The base Bridge, without additional modules/packages/libraries aims to incorporate basically .NET's coreclr, which builds into valid JavaScript code.

Maybe with your suggestion, the thought is like "have javascript written in C#" as you would with typescript?

**Comment by groogiam** [comment]
What I'm saying is that signature of this function does not match what it is in javascript. 

arr.indexOf(searchElement[, fromIndex])

The method provided on bridge only takes a string for the first parmater. This effectively makes it impossible to call indexOf on any array other than a string array without casting to dynamic. It seems like the indexOf should have the first parameter be object or dynamic.

**Comment by fabriciomurta** [comment]
The intended signature in Bridge is .NET's:
[String.IndexOf Method](https://docs.microsoft.com/en-us/dotnet/api/system.string.indexof?view=netframework-4.7.2).

And Bridge should output the code in JavaScript's.

Getting the JavaScript direct implementations can be achieved by either using libraries/extensions (like NuGet packages) or `Bridge.Script.Write()` to add raw lines of code.

To clarify, The base Bridge is .NET CLR implementation, that output JavaScript code. But sometimes javascript can be necessary or desired, so there's the two ways above.

At some point Bridge had the mixed signatures, and we had to make a choice between C# or JavaScript's. As Bridge is intended to "write C#, output javascript", we chose the former back in #386.

I'm not sure the [Bridge.Html5](https://www.nuget.org/packages/Bridge.Html5/) or [retyped.dom](https://www.nuget.org/packages/Retyped.dom/) NuGet packages, implement array extensions to give this signature to all arrays, but they add JavaScript's signatures to their implementations of typed arrays.
